### PR TITLE
chore: modernize tests

### PR DESCRIPTION
**Pull request by Maniktherana**
## Description
Used [test writer](https://github.com/arvind-murty/test-rewriter) to convert `t.errorf` to `assert`

## Related Issue(s)
addresses #15183
partially covers #15182

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by Maniktherana** [comment]
hey @ajm188 I tried to add the new range where possible in testfiles but on commiting linter's complaining with this output:
```
git commit -m "chore: use new range for loops"
2024/02/15 17:04:03 5 files OK
Linting go/bucketpool go/cache/theine go/mysql go/mysql/endtoend go/pools go/pools/smartconnpool
go/bucketpool/bucketpool_test.go:181:12: cannot range over 20000 (untyped int constant) (typecheck)
        for range 20000 {
                  ^
```
not sure if the issue's with CI or with my machine as I've been having some issues setting up go1.22

**Comment by ajm188** [comment]
> not sure if the issue's with CI or with my machine as I've been having some issues setting up go1.22

if you aren't on at least 1.22 this syntax is invalid, so that is probably the issue. what is the issue you are having setting up on your machine? i'm happy to help!

**Comment by Maniktherana** [comment]
I did install go1.22 and it's now the default go version on my machine. These errors don't show up in vscode. Only shows up when I try to commit via my terminal 🤔

As for my setup I'm having issues with the Go lsp on vscode but I think that's unrelated.

**Comment by ajm188** [comment]
I just tried applying that change and didn't have a problem:

```
$ git commit -sS -m 'test commit'
2024/02/15 07:58:42 5 files OK
Linting go/bucketpool
[andrew/delete-vtctl-commands 12a6beabaa] test commit
 1 file changed, 1 insertion(+), 1 deletion(-)
$ git show HEAD -p
commit 12a6beabaa1dfdff4e2470871d4725fcea8c9203
Author: Andrew Mason <andrew@planetscale.com>
Date:   Thu Feb 15 07:58:39 2024 -0500

    test commit
    
    Signed-off-by: Andrew Mason <andrew@planetscale.com>

diff --git a/go/bucketpool/bucketpool_test.go b/go/bucketpool/bucketpool_test.go
index cc8bcff0ee..6f088caddc 100644
--- a/go/bucketpool/bucketpool_test.go
+++ b/go/bucketpool/bucketpool_test.go
@@ -178,7 +178,7 @@ func TestPoolWeirdMaxSize(t *testing.T) {
 
 func TestFuzz(t *testing.T) {
 	maxTestSize := 16384
-	for i := 0; i < 20000; i++ {
+	for range 20000 {
 		minSize := rand.Intn(maxTestSize)
 		if minSize == 0 {
 			minSize = 1
```

maybe you need to update golangci-lint after the go upgrade?

**Comment by ajm188** [comment]
@Maniktherana i think you've overlapped with some work i did here: https://github.com/vitessio/vitess/pull/15194 you might want to rebase on that

**Comment by Maniktherana** [comment]
> @Maniktherana i think you've overlapped with some work i did here: #15194 you might want to rebase on that

done

File: go/cache/lru_cache_test.go
Comment (frouioui) [inline go/cache/lru_cache_test.go:34]: When using the `asset` I don't think it is useful to add the expectations in the message field. The method already prints the wanted value and the received value:
```
Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestInitialState
        	Messages:   	size = 0, want 1
```

This apply to all the files you modified in this PR

Comment (Maniktherana) [reply go/cache/lru_cache_test.go]: done, I'll wait on the the CI checks tho

File: go/cache/theine/singleflight_test.go
Comment (ajm188) [inline go/cache/theine/singleflight_test.go:48]: on your next pass you should ditch this extra line

**Review by ajm188 — COMMENTED** [review]
looking good, pending previous review

File: go/cache/theine/singleflight_test.go
Comment (ajm188) [inline go/cache/theine/singleflight_test.go:57]: why did we change from `ErrorIs` to `Equal` here?

File: go/cache/theine/singleflight_test.go
Comment (ajm188) [inline go/cache/theine/singleflight_test.go:58]: why not `assert.Empty`?

File: go/cache/theine/singleflight_test.go
Comment (ajm188) [inline go/cache/theine/singleflight_test.go:101]: `assert.Greater` and `assert.Lesser` might produce more useful error messages

File: go/event/event_test.go
Comment (ajm188) [inline go/event/event_test.go:61]: this will read just as `expected true, got false`, so we might want a more descriptive message here (i.e. "expected static listener to trigger")

Comment (ajm188) [reply go/event/event_test.go:61]: same applies to other `assert.True` and `assert.False` instances

File: go/cache/theine/singleflight_test.go
Comment (Maniktherana) [reply go/cache/theine/singleflight_test.go:101]: ah, these are mostly from using [test writer](https://github.com/arvind-murty/test-rewriter) tool you mentioned. I'll go back and fix the tests

File: go/cache/theine/singleflight_test.go
Comment (Maniktherana) [reply go/cache/theine/singleflight_test.go:57]: this may have been due to me using the test-rewriter and then resolving merge conflicts when rebasing. I'll revert this

File: go/cache/theine/singleflight_test.go
Comment (Maniktherana) [reply go/cache/theine/singleflight_test.go:58]: I'll revert this one too

File: go/cache/theine/singleflight_test.go
Comment (ajm188) [reply go/cache/theine/singleflight_test.go:101]: yeah that tool isn't perfect but it's definitely a start over doing this all by hand! just needs some touching up after

File: go/event/event_test.go
Comment (deepthi) [reply go/event/event_test.go:61]: Agreed. If you are up to doing a PR against the test rewriter tool, it will be good to try and preserve the custom messages.

File: go/streamlog/streamlog_test.go
Comment (deepthi) [inline go/streamlog/streamlog_test.go:318]: 👍

**Review by deepthi — COMMENTED** [review]
Nice work. I can approve once the current set of comments are addressed. I didn't see anything else that needs to change.

**Comment by Maniktherana** [comment]
I've refactored the tests @deepthi @ajm188 
this should cover it

**Review by ajm188 — APPROVED** [review]

File: go/event/event_test.go
Comment (deepthi) [inline go/event/event_test.go:230]: You missed this one, it should have a message passed in.

File: go/mysql/decimal/decimal_test.go
Comment (deepthi) [inline go/mysql/decimal/decimal_test.go:787]: Should fix the formatting. Comment should go on the line above the assertion.

**Review by deepthi — COMMENTED** [review]
I found a couple of small things that still need to fixed. Will give final approval after that.

File: go/event/event_test.go
Comment (Maniktherana) [reply go/event/event_test.go]: done

File: go/mysql/decimal/decimal_test.go
Comment (systay) [inline go/mysql/decimal/decimal_test.go:1000]: nitpick: `assert.Zero`

File: go/mysql/decimal/decimal_test.go
Comment (systay) [inline go/mysql/decimal/decimal_test.go:946]: nitpick: `assert.NotZero`

File: go/cache/lru_cache_test.go
Comment (Maniktherana) [reply go/cache/lru_cache_test.go]: dont

File: go/cache/lru_cache_test.go
Comment (ajm188) [inline go/cache/lru_cache_test.go:37]: why did we make these changes?

**Review by ajm188 — COMMENTED** [review]
i'm on-board with the `assert.True` messages and the Greater/Less, but i have questions about the Zero and EqualValues

**Review by deepthi — APPROVED** [review]

**Pull request merged**
